app/grandchallenge/core/forms.py

```diff
@@ -1,19 +1,48 @@
+from crispy_forms.bootstrap import StrictButton
 from crispy_forms.helper import FormHelper
-from crispy_forms.layout import Submit
+from crispy_forms.layout import Fieldset, Layout
 from django.conf import settings
 from django.core.exceptions import ValidationError
 from django.forms import CharField, ModelForm
+from django.templatetags.static import static
+from django.utils.html import format_html
 
 from grandchallenge.core.guardian import filter_by_permission
 from grandchallenge.workstation_configs.models import WorkstationConfig
 from grandchallenge.workstations.models import Workstation
 
 
 class SaveFormInitMixin:
+    """
+    Mixin that adds some save features to a form via init:
+      - a 'Save' button
+      - disabling fieldsets after the form is submitted
+    """
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.helper = FormHelper(self)
-        self.helper.layout.append(Submit("save", "Save"))
+        self.helper.attrs["gc-disable-after-submit"] = True
+        self.helper.layout = Layout(
+            Fieldset(
+                None,  # Legend
+                self.helper.layout,
+            ),
+            StrictButton(
+                "Save",
+                css_class="btn-primary",
+                type="submit",
+                css_id="submit-id-save",
+            ),
+        )
+
+    class Media:
+        js = [
+            format_html(
+                '<script type="module" src="{}"></script>',
+                static("js/disable_after_submit.mjs"),
+            )
```

Comment:
Why did you use this? You should be able to do the following, right?

```suggestion
            "js/disable_after_submit.mjs",
```